src/FakeItEasy/Core/ArgumentConstraintTrap.cs

```diff
@@ -11,10 +11,12 @@ internal class ArgumentConstraintTrap
 
         public static void ReportTrappedConstraint(IArgumentConstraint constraint)
         {
-            if (trappedConstraints != null)
+            if (trappedConstraints == null)
             {
-                trappedConstraints.Add(constraint);
+                throw new InvalidOperationException("A.Ignored, A._, and A.That can only be used in the context of a call specification with A.CallTo()");

```

Comment:
Reading this again, I realize it should be `A<T>.Ignored` rather than `A.Ignored`... makes the message harder to read, though. What do you think?